kratos/geometries/geometry_data.h

```diff
@@ -249,31 +262,61 @@ class GeometryData
     GeometryData( SizeType ThisDimension,
                   SizeType ThisWorkingSpaceDimension,
                   SizeType ThisLocalSpaceDimension,
-                  enum IntegrationMethod ThisDefaultMethod,
+                  IntegrationMethod ThisDefaultMethod,
                   const IntegrationPointsContainerType& ThisIntegrationPoints,
                   const ShapeFunctionsValuesContainerType& ThisShapeFunctionsValues,
                   const ShapeFunctionsLocalGradientsContainerType& ThisShapeFunctionsLocalGradients )
-        : mDimension( ThisDimension )
-        , mWorkingSpaceDimension( ThisWorkingSpaceDimension )
-        , mLocalSpaceDimension( ThisLocalSpaceDimension )
-        , mDefaultMethod( ThisDefaultMethod )
-        , mIntegrationPoints( ThisIntegrationPoints )
-        , mShapeFunctionsValues( ThisShapeFunctionsValues )
-        , mShapeFunctionsLocalGradients( ThisShapeFunctionsLocalGradients )
+        : mGeometryShapeFunctionContainer(
+            GeometryShapeFunctionContainer<IntegrationMethod>(
+                ThisDefaultMethod,
+                ThisIntegrationPoints,
+                ThisShapeFunctionsValues,
+                ThisShapeFunctionsLocalGradients))
+    {
+        mpGeometryDimension = new GeometryDimension(
```

Comment:
@pooyan-dadvand do you think this constructor could lead to memory problems. It is never destructed.

Reply:
It will.

Do we need this constructor? If yes, we should make the `mpGeometryDimension` a unique pointer.

Reply:
This constructor is needed for backwards compabibility.

Reply:
@pooyan-dadvand I changed the geometries that the new constructor is not needed anymore